### ISSUE TITLE
Add support for Dash

### ISSUE THREAD

**Issue by yivo** [openware/peatio#346]

**Comment by Vivek-Patil** [comment]
@yivo So I see that the RPC commands will be the same as for BTC, BCH, LTC ? I was looking into the dash documentation and found out that those commands are deprecated. https://dashpay.atlassian.net/wiki/spaces/DOC/pages/131924073/RPC+commands.

**Comment by yivo** [comment]
Hello @Vivek-Patil 

You can get the list of commands we are using at https://github.com/rubykube/peatio/blob/devel/lib/coin_api/btc.rb

Now we have no time to update the implementation to match new RPC.

**Comment by Vivek-Patil** [comment]
@yivo Thanks for your reply. I just wanted to make sure that these rpc commands work with the actual dash node hosted.Have you tested these on the latest dash node and it works right?

**Comment by yivo** [comment]
@Vivek-Patil Yes, it works with the latest Dashd.

**Comment by yivo** [comment]
@Vivek-Patil If you want to test check out http://peatio.trade:4000/

**Comment by Vivek-Patil** [comment]
@yivo Cool. One more question about Ripple.The current implementation of Ripple it seems that does not support the concept of tags.Also the current implemetation works with testnet which ripple has i.e https://ripple.com/build/xrp-test-net/. But the rest API's used here are not available even on the public ripple server or even the self hosted rippled.Also this implementation creates new wallet for each user which cant work with the exchange which is what i understand. I would like to know your thoughts on this if I am getting something wrong here.

**Comment by yivo** [comment]
@Vivek-Patil I am not sure if current Ripple implementation should supports destination tags. I know the issue with old v1 REST API, I am looking to put coin_api/xrp.rb back (I recently removed it due to integration with BitGo). I would like to fully migrate to Rippled RPC. I am not sure for your last question but now API returns address and secret for every new address which is stored in database. I am not very familiar with Ripple. BitGo has different implementation – it returns the same address but with new dt like address?dt=123

**Comment by Vivek-Patil** [comment]
@yivo I just check today that Ripple has even removed the rest api's now from testing environment, so this wont work anyways.I think this needs to be fully replaced with RPC implementation now.

**Comment by Vivek-Patil** [comment]
@yivo Can you please help me regarding the dash core. i am running this docker image for dashd https://github.com/dashpay/docker-dashd and when i try running getnewaddress on dash-cli it says command not found. What is version of dashd with which you have tested the code.?

**Comment by yivo** [comment]
@Vivek-Patil getinfo returns
```
{"result":{"version":120203,"protocolversion":70208,"walletversion":61000,"balance":566.92730970,"privatesend_balance":0.00000000,"blocks":69423,"timeoffset":-28,"connections":8,"proxy":"","difficulty":16.62762727940368,"testnet":true,"keypoololdest":1517427394,"keypoolsize":999,"paytxfee":0.00010000,"relayfee":0.00001000,"errors":""},"error":null,"id":null}
```

**Comment by Vivek-Patil** [comment]
@yivo Yes i got the same info for this command.But how does other commands work.Are they same as BTC, BCH, LTC node? I am running the command dash-cli getnewaddress and it gives error.But your above PR shows me that you are using the same RPC commands , but why am i getting error?

**Comment by yivo** [comment]
@Vivek-Patil I don't know. We tested Dash and it worked. Maybe you haven't enabled RPC server?

**Comment by Vivek-Patil** [comment]
@yivo Ohh yes, i was missing that. It works. Cheers !!

**Comment by Vivek-Patil** [comment]
@yivo Any plans on how to proceed with Ripple, as currenctly they have disabled the rest api's even on testnet permanently, so the current code wont even work :(

**Comment by yivo** [comment]
Closed in #425.
